autoria.py

```diff
@@ -213,41 +245,166 @@ def get_average(self, category: str, mark: str, model: str,
+        """
+        self._api = RiaAPI()
+        self._params = {}
 
         # Getting list of categories and selecting needed id
-        categories_list = self.get_categories()
-        selected_category = self.select_item(category, categories_list)
-        params_dict['main_category'] = selected_category
+        categories_list = self._api.get_categories()
+        selected_category = select_item(category, categories_list)
+        self._params['main_category'] = selected_category
 
         # Getting list of bodysyles and selecting needed id
-        bodystyles_list = self.get_bodystyles(selected_category)
-        selected_bodystyle = self.select_item(bodystyle, bodystyles_list)
-        params_dict['body_id'] = selected_bodystyle
+        bodystyles_list = self._api.get_bodystyles(selected_category)
+        self._params['body_id'] = select_item(bodystyle, bodystyles_list)
 
         # Getting list of marks and selecting needed id
-        marks_list = self.get_marks(selected_category)
-        selected_mark = self.select_item(mark, marks_list)
-        params_dict['marka_id'] = selected_mark
+        marks_list = self._api.get_marks(selected_category)
+        selected_mark = select_item(mark, marks_list)
+        self._params['marka_id'] = selected_mark
 
         # Getting list of models and selecting needed id
-        models_list = self.get_models(selected_category, selected_mark)
-        selected_model = self.select_item(model, models_list)
-        params_dict['model_id'] = selected_model
+        models_list = self._api.get_models(selected_category, selected_mark)
+        self._params['model_id'] = select_item(model, models_list)
 
         # Getting list of states and selecting needed id
-        states_list = self.get_states()
-        selected_state = self.select_item(state, states_list)
-        params_dict['state_id'] = selected_state
-
-        # Creting a list from start and end years
-        params_dict['yers'] = years
-
-        # Getting average price
-        url = 'http://api.auto.ria.com/average'
-        # Senging GET request
-        response = requests.get(url, params=params_dict)
-        # Formatiing response into JSON for convenience
-        average = json.loads(response.text)
-        print('{}\n{}'.format(params_dict, response.url))
-        return average
+        states_list = self._api.get_states()
+        self._params['state_id'] = select_item(state, states_list)
+
+        # Creating a list from start and end years
+        self._params['yers'] = years
+
+        # Adding a list of milage to params
+        self._params['raceInt'] = mileage
+
+        # Creating a list of gearbox ids
+        gearboxes = self._api.get_gearboxes(selected_category)
+        self._params['gear_id'] = select_list(gears, gearboxes)
+
+        # Creating a list of option ids
+        options = self._api.get_options(selected_category)
+        self._params['options'] = select_list(opts, options)
```

Comment:
Yeah, I was thinking about this way because it allows to maintain all parameters like in one place. And also I've made a type in params / _params like 10 times already :D I'll rework code this way, thanks.